scripts/amd/setup_rocm_libs.sh

```diff
@@ -0,0 +1,148 @@
+#!/usr/bin/env bash
+
+set -ex
+
+CWD=$(dirname "$0")
+
+echo "Collecting Necessary ROCM libraries"
+if [[ -z "${ROCM_VERSION}" ]]; then
+  ROCM_VERSION=5.4.2
+fi
+
+if [[ -z "${TRITON_ROCM_DIR}" ]]; then
+  TRITON_ROCM_DIR=python/triton/third_party/rocm
+fi
+  
+EXTRACT_DIR=/tmp/vanilla_extract
+mkdir -p $EXTRACT_DIR
+
+save_IFS="$IFS"
+IFS=. ROCM_VERSION_ARRAY=(${ROCM_VERSION})
+IFS="$save_IFS"
+if [[ ${#ROCM_VERSION_ARRAY[@]} == 2 ]]; then
+    ROCM_VERSION_MAJOR=${ROCM_VERSION_ARRAY[0]}
+    ROCM_VERSION_MINOR=${ROCM_VERSION_ARRAY[1]}
+    ROCM_VERSION_PATCH=0
+elif [[ ${#ROCM_VERSION_ARRAY[@]} == 3 ]]; then
+    ROCM_VERSION_MAJOR=${ROCM_VERSION_ARRAY[0]}
+    ROCM_VERSION_MINOR=${ROCM_VERSION_ARRAY[1]}
+    ROCM_VERSION_PATCH=${ROCM_VERSION_ARRAY[2]}
+else
+    echo "Unhandled ROCM_VERSION ${ROCM_VERSION}"
+    exit 1
+fi
+
+ROCM_INT=$(($ROCM_VERSION_MAJOR * 10000 + $ROCM_VERSION_MINOR * 100 + $ROCM_VERSION_PATCH))
+OS_NAME=`awk -F= '/^NAME/{print $2}' /etc/os-release`
+
+if [[ $ROCM_INT -eq 50402 ]]; then
+    if [[ "$OS_NAME" == *"CentOS Linux"* ]]; then
+        ROCM_PKGS=(
+            "https://repo.radeon.com/rocm/yum/5.4.2/main/hsa-rocr-1.7.0.50402-104.el7.x86_64.rpm"
+            "https://repo.radeon.com/rocm/yum/5.4.2/main/hip-runtime-amd-5.4.22803.50402-104.el7.x86_64.rpm"
+            "https://repo.radeon.com/rocm/yum/5.4.2/main/hip-devel-5.4.22803.50402-104.el7.x86_64.rpm"
+            "https://repo.radeon.com/rocm/yum/5.4.2/main/rocm-llvm-15.0.0.22506.50402-104.el7.x86_64.rpm"
+            "https://repo.radeon.com/rocm/yum/5.4.2/main/comgr-2.4.0.50402-104.el7.x86_64.rpm"
+        )
+
```

Comment:
From Pruthvi..would it be worth just unpacking centos7 version of rocm, regardless of operating system.

Reply:
It's a fair question: unpacking full rocm instead of individual packages might make this file a bit cleaner and easier to update  for future ROCm versions, at the cost of spending more time downloading a lot of unnecessary packages. 
As for unpacking centos7 version of rocm, regardless of operating system, that might be okay if the tools used to unpack the packages in each OS ([dpkg ](https://github.com/ROCmSoftwarePlatform/triton/pull/195/files/8f705101e5366f6b21895e85bfb38e0870e7cd59#diff-5de9ed6679de1be0f67165f238efe66c2d716c99aca32631a3021c2e93793207R100) vs [rpm2cpio/cpio](https://github.com/ROCmSoftwarePlatform/triton/pull/195/files/8f705101e5366f6b21895e85bfb38e0870e7cd59#diff-5de9ed6679de1be0f67165f238efe66c2d716c99aca32631a3021c2e93793207R73)) are available in both OS environments, especially since this code seems like it handles the subtle directory structure difference b/w Centos and Ubuntu packages.

Reply:
are we okay if we leave this in for now and address in a next release?

Reply:
Absolutely, this is in "refactor" category.